libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -781,7 +781,9 @@ bool AP_Arming::system_checks(bool report)
 #endif
     }
     if (AP::internalerror().errors() != 0) {
-        check_failed(report, "Internal errors (0x%x) (last line:%u)", (unsigned int)AP::internalerror().errors(), AP::internalerror().last_error_line());
+        uint8_t buffer[32]{};
```

Comment:
Why have an initializer here? Seems like a waste of processor time/code. `errors_as_string` is filling it in, and you are sending the whole thing into the check_failed format string which will stop at the first null termination.

Reply:
I considered this - it was basically a defensive thing - in case `errors_as_string` didn't do the right thing, basically.

Processor time - not exactly an issue considering where this is.
Code size?  I'd guess a couple of dozen bytes worst-case.

Reply:
I had the same thought on speed, but size sorta still seemed relevant. I guess I'm just a bit hesitant to encourage people to always init stuff they don't need :)

Reply:
I've removed that initialisation and guaranteed the returned buffer is null-terminated